addons/html_editor/static/src/utils/dom_traversal.js

```diff
@@ -58,6 +58,22 @@ export function closestElement(node, predicate = "*") {
     return null;
 }
 
+/**
+ * @param {Node} node
+ * @param {HTMLElement} limitAncestor - non inclusive limit ancestor to search for
+ * @param {Function} predicate
+ * @returns {Node|null}
+ */
+export function closestNode(node, limitAncestor, predicate) {
```

Comment:
It's all about the non-inclusive `limitAncestor`. It`s meant to restrict the search to a specific subtree (in the DeletePlugin, it is usually the subtree rooted at the `commonAncestorContainer`)

I guess I could use `closestElement` like this:
```js
const closestEl = closestElement(node, (n) => limitAncestor.contains(n) && n !== limitAncestor && myPredicate(node)
```

instead of 
```js
const closestEl = closestNode(node, myPredicate)
```

Other than having to add logic to check if it is contained under the desired subtree in the predicate function, it is much less efficient (when there's no element to be found, it will keep searching until the editable root even though it's past the limit ancestor).

Not to mention this function is much more simple and readable...